download-mksnapshot.js

```diff
@@ -5,15 +5,30 @@ const extractZip = require('extract-zip')
 const versionToDownload = require('./package').version
 let archToDownload = process.env.npm_config_arch
 
-if (process.arch.indexOf('arm') === 0 && process.platform !== 'darwin') {
-  console.log(`WARNING: mksnapshot does not run on ${process.arch}. Download 
-  https://github.com/electron/electron/releases/download/v${versionToDownload}/mksnapshot-v${versionToDownload}-${process.platform}-${process.arch}-x64.zip
-  on a x64 ${process.platform} OS to generate ${archToDownload} snapshots.`)
-  process.exit(1)
+// From electron/fiddle-core
+function compareVersions (a, b) {
+  const l = a.compareMain(b)
+  if (l) return l
+  // Electron's approach is nightly -> other prerelease tags -> stable,
+  // so force `nightly` to sort before other prerelease tags.
+  const [prea] = a.prerelease
+  const [preb] = b.prerelease
+  if (prea === 'nightly' && preb !== 'nightly') return -1
+  if (prea !== 'nightly' && preb === 'nightly') return 1
+  return a.comparePre(b)
 }
 
-if (archToDownload && archToDownload.indexOf('arm') === 0 && process.platform !== 'darwin') {
-  archToDownload += '-x64'
+function getLatestStable (releases) {
+  const { parse: semverParse } = require('semver')
+  const semvers = releases.map(({ version }) => semverParse(version)).filter((sem) => Boolean(sem))
+  semvers.sort((a, b) => compareVersions(a, b))
+  let stable
+  for (const ver of semvers.values()) {
+    if (ver.prerelease.length === 0) {
+      stable = ver
```

Comment:
Just a sanity check here - does `sort` do asc or desc by default? Want to make sure we're returning the most recent version

Reply:
This is a good call out, I was also confused by this initially. This is all ripped from `electron/fiddle-core`, I'll put better comments reffing the spots I pulled the code from.